app/helpers/react_on_rails_helper.rb

```diff
@@ -47,53 +56,97 @@ def react_component(component_name, props = {}, options = {})
     data_from_server_script_tag = javascript_tag(page_loaded_js)
 
     # Create the HTML rendering part
+    server_rendered_html =
+      server_rendered_react_component_html(options, props, react_component_name)
+
+    rendered_output = content_tag(:div,
+                                  server_rendered_html,
+                                  id: dom_id)
+
+    <<-HTML.strip_heredoc.html_safe
+#{data_from_server_script_tag}
+    #{rendered_output}
+    HTML
+  end
+
+  def next_react_component_index
+    @react_component_index ||= -1
+    @react_component_index += 1
```

Comment:
@samnang the instance variables work very well as Rails includes the helpers in the view object. I don't see where else we would put this. Note, there's no threading issues here. The `@` variables are like other class instance variables.